examples/debugger/direct.c

```diff
@@ -358,6 +436,23 @@ static int cospawn_launch(myrel_t *myrel)
     DEBUG_WAIT_THREAD(&mylock);
     DEBUG_DESTRUCT_LOCK(&mylock);
     PMIX_INFO_FREE(info, 2);
+    /* We need to issue PMIx_Fence here to satisfy the corresponding fence in the
```

Comment:
At first, I was really puzzled by what this is doing - why would we want the tool to get slurped up into the daemon job? I then saw that you were extending the fence to include this proc plus the daemon job - i.e., this tool isn't becoming part of the daemon job, just executing a fence with it.

I suppose that can take the place of registering for a "launch-complete" or "job-running" event, though it does seem a little weird (or at least unexpected given that it differs from how we handle the other events). I can't offhand think of any secondary consequences of this approach - probably worth pondering a bit more.

However, I'm not sure how it correlates with the _app_ reaching "debug-ready" - are you assuming the daemons will "trap" the local "debug-ready" event and only enter the fence once receiving it? I don't think that will be true in general, so might be worth explaining that assumption somewhere.

Reply:
cross-namespace fence is working - or else MPI comm_spawn would fail. The root problem seems to be the error in having the app and daemon in the same namespace. Rather than trying to devise workarounds to make the test "run", it would probably be better to spend that time fixing the root cause. There seems little value in having a test that works with a broken library :-/

Reply:
We agree. Dave just pushed the `#if 0` to block out the cospawn case for now. We'll work on the root cause of cospawn then return to fix up that case in the example.